In [10]:
import pandas as pd
df = pd.read_csv(r'C:\Users\Admin\PROJECT\Project_instacart\data\orders_with_dates_2025.csv')

In [11]:
print(f"\nTotal rows: {len(df)}")
print("\nFirst few rows:")
print(df.head(2))

print("\nData types:")
print(df.dtypes)



Total rows: 3000

First few rows:
   order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   

   days_since_prior_order  order_date  
0                     NaN  2025-12-11  
1                    15.0  2025-11-24  

Data types:
order_id                    int64
user_id                     int64
eval_set                   object
order_number                int64
order_dow                   int64
order_hour_of_day           int64
days_since_prior_order    float64
order_date                 object
dtype: object


In [12]:
print("\nBasic statistics:")
print(df.describe())


Basic statistics:
           order_id      user_id  order_number    order_dow  \
count  3.000000e+03  3000.000000   3000.000000  3000.000000   
mean   1.758135e+06   102.373667     14.962333     2.750667   
std    9.896337e+05    59.421772     16.043671     2.047412   
min    1.483000e+03     1.000000      1.000000     0.000000   
25%    9.317088e+05    51.000000      4.000000     1.000000   
50%    1.765432e+06    99.000000      9.000000     3.000000   
75%    2.631220e+06   150.000000     19.000000     5.000000   
max    3.415043e+06   248.000000     98.000000     6.000000   

       order_hour_of_day  days_since_prior_order  
count        3000.000000             2799.000000  
mean           13.190000               11.584852  
std             4.082253                9.196559  
min             0.000000                0.000000  
25%            10.000000                5.000000  
50%            13.000000                8.000000  
75%            16.000000               16.000000  
max  

In [13]:
# change order_date to datetime
df['order_date'] = pd.to_datetime(df['order_date'])

In [14]:
# check for order_date range
print(df['order_date'].min(), df['order_date'].max())

2025-11-10 00:00:00 2026-01-08 00:00:00


In [15]:
# checj order_id uniqueness
unique_order_ids = df['order_id'].nunique()
print(f"Unique order IDs: {unique_order_ids}")

Unique order IDs: 3000


## Columns:

- order_id - unique identifier
- user_id - customer identifier
- eval_set - whether this is prior/train/test data
- order_number - which order this is for the customer (1st, 2nd, 3rd...)
- order_dow - day of week (0-6, where 0=Sunday)
- order_hour_of_day - hour (0-23)
- days_since_prior_order - days since last order (NaN for first orders)

## Key facts:

- 3.4 million orders
- ~206k unique users
- Orders span all days of week and all hours